src/compile-route.js

```diff
@@ -27,16 +28,48 @@ function getHeaderMatcher (expectedHeaders) {
 		if (!headers) {
 			headers = {};
 		}
+
+		if (headers instanceof Headers) {
+			headers = headers.raw();
+		}
+
 		const lowerCaseHeaders = Object.keys(headers).reduce((obj, k) => {
 			obj[k.toLowerCase()] = headers[k]
 			return obj;
 		}, {});
+
 		return expectation.every(header => {
-			return lowerCaseHeaders[header.key] === header.val;
+			return areHeadersEqual(lowerCaseHeaders, header);
 		})
 	}
 }
 
+function areHeadersEqual (currentHeader, expectedHeader) {
+    const {key, val} = expectedHeader;
+    const currentHeaderValue = (Array.isArray(currentHeader[key])) ? currentHeader[key] : [currentHeader[key]];
+    const expectedHeaderValue = (Array.isArray(val)) ? val : [val];
+
+    if (currentHeaderValue === expectedHeaderValue) {
+		return true;
+    }
+
+    if (!currentHeaderValue || !expectedHeaderValue) {
```

Comment:
I just tried to be as much error-prone as possible. You're correct - I cannot think of a way of having an invalid `expectedHeaderValue` in here.

In fact, because of wrapping both incoming params into an array, the usage of `(currentHeaderValue.length !== expectedHeaderValue.length)` should be enough, shouldn't it?